homeassistant/components/combined_energy/sensor.py

```diff
@@ -0,0 +1,529 @@
+        ),
+        SensorEntityDescription(
+            key="power_factor_b",
+            name="Power Factor B",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="%",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="power_factor_c",
+            name="Power Factor C",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="%",
+            device_class=SensorDeviceClass.POWER_FACTOR,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="voltage_a",
+            name="Voltage A",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+        ),
+        SensorEntityDescription(
+            key="voltage_b",
+            name="Voltage B",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+        SensorEntityDescription(
+            key="voltage_c",
+            name="Voltage C",
+            state_class=SensorStateClass.MEASUREMENT,
+            native_unit_of_measurement="V",
+            device_class=SensorDeviceClass.VOLTAGE,
+            entity_registry_enabled_default=False,
+        ),
+    ],
+    "GENERIC_CONSUMER": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+    "ENERGY_BALANCE": SENSOR_DESCRIPTIONS_GENERIC_CONSUMER,
+}
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up sensors."""
+
+    api: CombinedEnergy = hass.data[DOMAIN][entry.entry_id][DATA_API_CLIENT]
+    installation: Installation = hass.data[DOMAIN][entry.entry_id][DATA_INSTALLATION]
+
+    # Initialise services
+    log_session = CombinedEnergyLogSessionService(hass, api)
+    log_session.async_setup()
+    await log_session.coordinator.async_refresh()
+
+    readings = CombinedEnergyReadingsDataService(hass, api)
+    readings.async_setup()
+    await readings.coordinator.async_refresh()
+
+    # Store log session into Data
+    hass.data[DOMAIN][entry.entry_id][DATA_LOG_SESSION] = log_session
+
+    # Build entity list
+    sensor_factory = CombinedEnergyReadingsSensorFactory(hass, installation, readings)
+    entities: list[CombinedEnergyReadingsSensor] = list(sensor_factory.entities())
+    async_add_entities(entities)
+
+
+class CombinedEnergyReadingsSensor(CoordinatorEntity, SensorEntity):
+    """Representation of a Combined Energy API reading energy sensor."""
+
+    data_service: CombinedEnergyReadingsDataService
+    entity_description: SensorEntityDescription
+
+    native_value_rounding: int = 2
```

Comment:
Please use [`suggested_display_precision`](https://developers.home-assistant.io/docs/core/entity/sensor) instead so the user can change it if needed.